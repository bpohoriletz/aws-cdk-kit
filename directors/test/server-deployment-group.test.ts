import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import ServerDeploymentGroupDirector from '../server-deployment-group';
import Ec2DeploymentGroupBuilder from '../../codedeploy/server-deployment-group-builders/ec2-deployment-group-builder';
import { stub } from '../../test/stubs';

describe('new ServerDeploymentGroupDirector()', () => {
  let instance: ServerDeploymentGroupDirector;
  let stack: cdk.Stack;

  beforeEach(() => {
    instance = new ServerDeploymentGroupDirector(Ec2DeploymentGroupBuilder);
    stack = new cdk.Stack();
  });

  describe('#constructEc2Group', () => {
    it('uses defaults if only name is passed', () => {
      instance.setResources({ name: 'test-cd-dg' }).constructEc2Group(stack, 'DeploymentGroup');

      expect(Template.fromStack(stack).toJSON().Resources).toMatchSnapshot();
    });

    it('can create groups for the same application', () => {
      instance.setResources({ name: 'test-cd-dg', application: stub(stack, 'cd.ServerApplication') });
      instance.constructEc2Group(stack, 'DeploymentGroupStaging');
      instance.constructEc2Group(stack, 'DeploymentGroupDevelop');

      expect(Template.fromStack(stack).toJSON().Resources).toMatchSnapshot();
    });
  });
});
