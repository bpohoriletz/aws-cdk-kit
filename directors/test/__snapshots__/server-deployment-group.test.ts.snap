// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new ServerDeploymentGroupDirector() #constructAsgGroup uses defaults if only name is passed 1`] = `
{
  "DeploymentGroup6D277AF0": {
    "Properties": {
      "ApplicationName": {
        "Ref": "DeploymentGroupApplication7B89ABEB",
      },
      "AutoRollbackConfiguration": {
        "Enabled": true,
        "Events": [
          "DEPLOYMENT_FAILURE",
        ],
      },
      "DeploymentConfigName": "CodeDeployDefault.AllAtOnce",
      "DeploymentGroupName": "test-cd-dg",
      "ServiceRoleArn": {
        "Fn::GetAtt": [
          "DeploymentGroupRoleC40B98DA",
          "Arn",
        ],
      },
    },
    "Type": "AWS::CodeDeploy::DeploymentGroup",
  },
  "DeploymentGroupApplication7B89ABEB": {
    "Properties": {
      "ComputePlatform": "Server",
    },
    "Type": "AWS::CodeDeploy::Application",
  },
  "DeploymentGroupRoleC40B98DA": {
    "Properties": {
      "AssumeRolePolicyDocument": {
        "Statement": [
          {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": {
              "Service": "codedeploy.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "ManagedPolicyArns": [
        {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":iam::aws:policy/service-role/AWSCodeDeployRole",
            ],
          ],
        },
      ],
    },
    "Type": "AWS::IAM::Role",
  },
}
`;

exports[`new ServerDeploymentGroupDirector() #constructEc2Group can create groups for the same application 1`] = `
{
  "DeploymentGroupDevelop5C995268": {
    "Properties": {
      "ApplicationName": {
        "Ref": "DeploymentGroupDevelopApplication8497067F",
      },
      "AutoRollbackConfiguration": {
        "Enabled": true,
        "Events": [
          "DEPLOYMENT_FAILURE",
        ],
      },
      "DeploymentConfigName": "CodeDeployDefault.AllAtOnce",
      "DeploymentGroupName": "test-cd-dg-devel",
      "ServiceRoleArn": {
        "Fn::GetAtt": [
          "DeploymentGroupDevelopRole3F2B6EFE",
          "Arn",
        ],
      },
    },
    "Type": "AWS::CodeDeploy::DeploymentGroup",
  },
  "DeploymentGroupDevelopApplication8497067F": {
    "Properties": {
      "ComputePlatform": "Server",
    },
    "Type": "AWS::CodeDeploy::Application",
  },
  "DeploymentGroupDevelopRole3F2B6EFE": {
    "Properties": {
      "AssumeRolePolicyDocument": {
        "Statement": [
          {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": {
              "Service": "codedeploy.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "ManagedPolicyArns": [
        {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":iam::aws:policy/service-role/AWSCodeDeployRole",
            ],
          ],
        },
      ],
    },
    "Type": "AWS::IAM::Role",
  },
  "DeploymentGroupStaging19AF76AC": {
    "Properties": {
      "ApplicationName": {
        "Ref": "DeploymentGroupStagingApplication50335847",
      },
      "AutoRollbackConfiguration": {
        "Enabled": true,
        "Events": [
          "DEPLOYMENT_FAILURE",
        ],
      },
      "DeploymentConfigName": "CodeDeployDefault.AllAtOnce",
      "DeploymentGroupName": "test-cd-dg-stag",
      "ServiceRoleArn": {
        "Fn::GetAtt": [
          "DeploymentGroupStagingRole5DBB2C01",
          "Arn",
        ],
      },
    },
    "Type": "AWS::CodeDeploy::DeploymentGroup",
  },
  "DeploymentGroupStagingApplication50335847": {
    "Properties": {
      "ComputePlatform": "Server",
    },
    "Type": "AWS::CodeDeploy::Application",
  },
  "DeploymentGroupStagingRole5DBB2C01": {
    "Properties": {
      "AssumeRolePolicyDocument": {
        "Statement": [
          {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": {
              "Service": "codedeploy.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "ManagedPolicyArns": [
        {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":iam::aws:policy/service-role/AWSCodeDeployRole",
            ],
          ],
        },
      ],
    },
    "Type": "AWS::IAM::Role",
  },
  "ServerApplicationID0A7B7E9A": {
    "Properties": {
      "ApplicationName": "cd.ServerApplication",
      "ComputePlatform": "Server",
    },
    "Type": "AWS::CodeDeploy::Application",
  },
}
`;

exports[`new ServerDeploymentGroupDirector() #constructEc2Group uses defaults if only name is passed 1`] = `
{
  "DeploymentGroup6D277AF0": {
    "Properties": {
      "ApplicationName": {
        "Ref": "DeploymentGroupApplication7B89ABEB",
      },
      "AutoRollbackConfiguration": {
        "Enabled": true,
        "Events": [
          "DEPLOYMENT_FAILURE",
        ],
      },
      "DeploymentConfigName": "CodeDeployDefault.AllAtOnce",
      "DeploymentGroupName": "test-cd-dg",
      "ServiceRoleArn": {
        "Fn::GetAtt": [
          "DeploymentGroupRoleC40B98DA",
          "Arn",
        ],
      },
    },
    "Type": "AWS::CodeDeploy::DeploymentGroup",
  },
  "DeploymentGroupApplication7B89ABEB": {
    "Properties": {
      "ComputePlatform": "Server",
    },
    "Type": "AWS::CodeDeploy::Application",
  },
  "DeploymentGroupRoleC40B98DA": {
    "Properties": {
      "AssumeRolePolicyDocument": {
        "Statement": [
          {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": {
              "Service": "codedeploy.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "ManagedPolicyArns": [
        {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":iam::aws:policy/service-role/AWSCodeDeployRole",
            ],
          ],
        },
      ],
    },
    "Type": "AWS::IAM::Role",
  },
}
`;
